### PR TITLE
Add Support for Parameter Files

### PR DESCRIPTION
**Pull request by jszaday**
This PR seamlessly adds parameter files alongside the existing scheme of command-line arguments, giving users the options to pass in a ChaNGa-style parameter file using the command-line option `-x` (e.g., `-x test.param`).

An example file is something like:
```
# comment...
nSubtreesMin = 48
nPartitionsMin = 48
nParticlesPerLeafMax = 24
achDecompType = oct
achTreeType = oct
```

~~**TODO:** Add a warning message if we get an unrecognized option.~~

### CODE REVIEW

**Comment by mojoe12** [comment]
@jszaday looks good! two questions though. one, can you make a github test for this at some point? second, if I do -x p_file -p 50, will the -p 50 override the p_file's value or vice versa?

**Comment by jszaday** [comment]
> @jszaday looks good! two questions though. one, can you make a github test for this at some point? second, if I do -x p_file -p 50, will the -p 50 override the p_file's value or vice versa?

Depends on the ordering. If the `-p` comes after the `-x` then it'll override it, otherwise, if the `-x` comes after the `-p` it'll override the `-p`. I wasn't sure how to address this matter, so left its behavior semi-undefined. We could do a pre-pass for `-x` and ignore everything else if it's found?

File: src/Loadable.C [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      auto& key = m[1];
+      auto& val = m[2];
+      // determine the type from the key
+      auto ty = Value::get_type(key);
+      // emplace the value with the appropriate conversion
+      auto ins = ([&](void) {
+        using T = Value::type;
+        switch (ty) {
+          case T::kBool:
+            return map.emplace(key, (bool)std::stoi(val));
```
Comment (trquinn) [inline src/Loadable.C:34]: I'm getting a compile error here with gcc v4.8.5 20150623 (Red Hat 4.8.5-44)
Loadable.C:34:57:   required from here
/usr/include/c++/4.8.2/bits/stl_pair.h:134:45: error: use of deleted function ‘paratreet::Value::Value(const paratreet::Value&)’
  : first(std::forward<_U1>(__x)), second(__y) { }
                                             ^
In file included from Field.hpp:4:0,
                 from Loadable.h:4,
                 from Loadable.C:1:
Value.hpp:54:3: error: declared here
   Value(const Value&) = delete;
   ^

**Review by trquinn — CHANGES_REQUESTED** [review]
Value.hpp and Field.hpp file names: we should agree on file name conventions; so far we have been using ".h" for C++ headers.
Also see comment at Loadable.C line 34

**Comment by jszaday** [comment]
I added a wiki page that covers the essentials for the new API additions:
https://github.com/paratreet/paratreet/wiki/Configuration-Customization

**Pull request merged**